lessons/207/http.zig/src/main.zig

```diff
@@ -0,0 +1,57 @@
+const std = @import("std");
+const httpz = @import("httpz");
+const Device = @import("device.zig");
+
+const port = 8080;
+
+pub fn main() !void {
+    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
+    const allocator = gpa.allocator();
+
+    var app = App{};
+
+    // playing around with fine tuning params to match what the prod system looks like
+    // with is 2 vCPUs ?
+    //
+    // This should consume around 2MB memory in total for sustained hits
+    var server = try httpz.Server(*App).init(allocator, .{
+        .address = "0.0.0.0",
+        .port = port,
+        .workers = .{
+            .large_buffer_count = 1,
+            .large_buffer_size = 1024,
+        },
+        .thread_pool = .{
+            .count = 1,
+            .buffer_size = 1024,
+        },
+        .request = .{
+            .buffer_size = 1024,
+            .max_header_count = 0,
+            .max_param_count = 0,
+            .max_query_count = 0,
+            .max_form_count = 0,
+            .max_multiform_count = 0,
+        },
+    }, &app);
+    var router = server.router(.{});
+    router.get("/healthz", App.healthz, .{});
+    router.get("/api/devices", App.getDevices, .{});
+
+    std.debug.print("http.zig listening on http://0.0.0.0:{d}\n", .{port});
+    try server.listen();
+}
+
```

Comment:
Note - App is a struct, and we created an instance of this in main, and wired our server up to it

All requests then get a handle to the App instance. You can then do things like have a DB object in the App instance, and other state information like that - so all requests can easily that App state

You can also have multiple different Objects instantiated, and wire different endpoints to different instances

Its pretty similar to Go from that regard ... and I think ... reasonably easy to follow for anyone familiar with Go

The middleware API is similar to Go as well